mentat/utils/path.py

```diff
@@ -0,0 +1,153 @@
+import fnmatch
+import re
+import os
+from pathlib import Path
+from typing import Iterable, Set, List
+
+from mentat.git_handler import check_is_git_repo, get_non_gitignored_files
+from mentat.errors import PathValidationException
+
+
+# TODO: replace this with something that doesn't load the file into memory
+def is_file_text_encoded(abs_path: Path):
+    """Checks if a file is text encoded."""
+    try:
+        # The ultimate filetype test
+        with open(abs_path, "r") as f:
+            f.read()
+        return True
+    except UnicodeDecodeError:
+        return False
+
+
+def validate_and_format_path(path: Path, cwd: Path, check_for_text: bool = True) -> Path:
+    """Validate and format a path.
+
+    `path` can be a file path, file interval path, directory path, or a glob pattern.
+    """
+    # Get absolute path
+    if path.is_absolute():
+        abs_path = path
+    else:
+        abs_path = cwd.joinpath(path)
+
+    # Validate different path types
+    # File
+    if abs_path.is_file():
+        if not abs_path.exists():
+            raise PathValidationException(f"File {abs_path} does not exist")
+        if check_for_text and not is_file_text_encoded(abs_path):
+            raise PathValidationException(f"Unable to read file {abs_path}")
+    # File interval
+    elif ":" in str(abs_path):
+        _interval_path, interval_str = str(abs_path).split(":", 1)
+        interval_path = Path(_interval_path)
+        if not interval_path.exists():
+            raise PathValidationException(f"File interval {abs_path} does not exist")
+        if check_for_text and not is_file_text_encoded(interval_path):
+            raise PathValidationException(f"Unable to read file interval {abs_path}")
+        abs_path = Path(f"{interval_path}:{interval_str}")
+    # Directory
+    elif abs_path.is_dir():
+        if not abs_path.exists():
+            raise PathValidationException(f"Directory {abs_path} does not exist")
+    # Glob pattern
+    elif re.search(r"[\*\?\[\]]", str(path)):
+        pass
+    else:
+        raise PathValidationException(f"Unable to validate path {path}")
+
+    return abs_path
+
+
+def match_path_with_patterns(path: Path, patterns: Set[str]) -> bool:
+    """Check if the given absolute path matches any of the patterns.
+
+    TODO: enforce valid glob patterns? Right now we allow glob-like
+    patterns (the one's that .gitignore uses). This feels weird imo.
```

Comment:
@biobootloader do you have any thoughts on this?

Reply:
I don't think we need this function; can't we stick with glob.glob like we were before?